packages/shared/src/components/streak/ReadingStreakButton.tsx

```diff
@@ -71,11 +87,11 @@ export function ReadingStreakButton({
   className,
 }: ReadingStreakButtonProps): ReactElement {
   const { logEvent } = useLogContext();
+  const { user } = useAuthContext();
   const isLaptop = useViewSize(ViewSize.Laptop);
   const isMobile = useViewSize(ViewSize.MobileL);
   const [shouldShowStreaks, setShouldShowStreaks] = useState(false);
-  const hasReadToday =
-    new Date(streak?.lastViewAt).getDate() === new Date().getDate();
+  const hasReadToday = getHasReadToday(streak?.lastViewAt, user?.timezone);
```

Comment:
But shouldn't API just return it in the correct format already?

Reply:
Currently, what is returned is the stored value, which is UTC.

Reply:
We can maybe add a prop where the returned value involves the right tz.

Reply:
So then we don't need this one?

Reply:
For backwards compatibility, I have kept the original column, and introduced a new one with timezone considered. So we'd still need this.